### PR TITLE
Master to PostgreSQL version 15

### PR DESCRIPTION
**Pull request by jrgemignani**
** **PLEASE DO NOT MERGE** ** until thoroughly reviewed and approved by all reviewers. Reviewers must state their approval as a comment in the thread below. Additionally, this merge needs to be appropriately squashed before merging.

PR for moving the master branch up to PostgreSQL version 15.

It contains all of the latest work in the master branch in addition to all of the changes made to go from PostgreSQL version 14 to 15. When this PR is merged the master will be the latest and on PostgreSQL version 15.

Everyone listed as a reviewer needs to review this PR

### CODE REVIEW

**Review by MuhammadTahaNaveed — APPROVED** [review]
Looks good.

**Comment by dehowef** [comment]
Looks good to me.

**Review by dehowef — APPROVED** [review]

File: src/backend/parser/cypher_gram.y [hunk 1]
```diff
@@ -2008,29 +2010,37 @@ static Node *do_negate(Node *n, int location)
...
         }
     }
 
     return (Node *)makeSimpleA_Expr(AEXPR_OP, "-", NULL, n, location);
 }
 
-static void do_negate_float(Value *v)
+static void do_negate_float(Float *v)
 {
-    Assert(IsA(v, Float));
```
Comment (rafsun42) [inline src/backend/parser/cypher_gram.y:2028]: Is removal of `Assert` intentional?

Comment (jrgemignani) [reply src/backend/parser/cypher_gram.y:2028]: I added it back in and some slight modifications.

**Review by muhammadshoaib — APPROVED** [review]

**Review by rafsun42 — APPROVED** [review]
Looks good.

**Pull request merged**